react-components/src/components/Reveal3DResources/Reveal3DResources.tsx

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { useRef, type ReactElement, useContext, useState, useEffect } from 'react';
+import { type Cognite3DViewer } from '@cognite/reveal';
+import { ModelsLoadingStateContext } from './ModelsLoadingContext';
+import CadModelContainer from '../CadModelContainer/CadModelContainer';
+import PointCloudContainer from '../PointCloudContainer/PointCloudContainer';
+import { Image360CollectionContainer } from '../Image360CollectionContainer/Image360CollectionContainer';
+import { useReveal } from '../RevealContainer/RevealContext';
+import {
+  type AddReveal3DModelOptions,
+  type AddImageCollection360Options,
+  type TypedReveal3DModel,
+  type AddResourceOptions
+} from './types';
+
+export type Reveal3DResourcesProps = {
+  resources: AddResourceOptions[];
+};
+
+export const Reveal3DResources = ({ resources }: Reveal3DResourcesProps): ReactElement => {
+  const [reveal3DModels, setReveal3DModels] = useState<TypedReveal3DModel[]>([]);
+  const { setModelsAdded } = useContext(ModelsLoadingStateContext);
+  const viewer = useReveal();
+  const numModelsLoaded = useRef(0);
+
+  useEffect(() => {
+    getTypedModels(resources, viewer).then(setReveal3DModels).catch(console.error);
+  }, []);
```

Comment:
Shouldn't this recompute if we change models parameters like id/revision or just add another model? I think adding `resources` as a dependency should be correct.

Reply:
I'll add it, but note that this doesn't come without down sides. F.ex. mutating the resources list / object doesn't actually cause this useeffect to run (comparrison by ref. vs. value).